src/sentry/models/projectownership.py

```diff
@@ -313,37 +315,40 @@ def handle_auto_assignment(cls, project_id, event=None, group=None):
                     )
                     return
 
-            assignment = GroupAssignee.objects.assign(
-                group,
-                owner,
-                create_only=not force_autoassign,
-                extra=details,
-                force_autoassign=force_autoassign,
-            )
-
-            if assignment["new_assignment"] or assignment["updated_assignment"]:
-                analytics.record(
-                    "codeowners.assignment"
-                    if details.get("integration") == ActivityIntegration.CODEOWNERS.value
-                    else "issueowners.assignment",
-                    organization_id=ownership.project.organization_id,
-                    project_id=project_id,
-                    group_id=group.id,
-                )
-                logger.info(
-                    "handle_auto_assignment.success",
-                    extra={
-                        "event": event.event_id if event else None,
-                        "group": group.id,
-                        "project": group.project.id,
-                        "organization": group.project.organization_id,
-                        # owner_id returns a string including the owner type (user or team) and id
-                        "assignee": issue_owner.owner_id(),
-                        "reason": "created" if assignment["new_assignment"] else "updated",
-                        **details,
-                    },
+            if not GroupAssignee.objects.filter(
+                Q(user_id=owner.id) | Q(team=owner.id), group=group
```

Comment:
Instead of checking both of these, shouldn't we check `owner.type` and apply either a user id or team id filter, rather than both? Probably the chance of ids overlapping is low, but still might be safer

Reply:
ahh yeah good catch